### ISSUE TITLE
[REQUEST] Add RNAlysis: A python based tool with gui for analysis of RNA sequencing data

### ISSUE THREAD

**Issue by kbipinkumar** [BioArchLinux/Packages#107]
**Information**
 - Package Name: RNAlysis
 - Can it directly run on Linux: yes
 - Can it run on the wine: Not applicable
 -  Website: https://github.com/GuyTeichman/RNAlysis
 - Download URL: https://github.com/GuyTeichman/RNAlysis/releases
 - Package description: RNAlysis is a Python-based software for analyzing RNA sequencing data with GUI.
 - DOI: 10.1101/2022.11.25.517851
 

**Popularity**
 - Any other reason: the tool is extremely user friendly despite being in preprint. it would be great addition to the repo as well.

**Additional information**

The tool makes extensive usage of many opensource packages. hence it has pretty long dependency list. some of them are available in main repo of archlinux. however those present in AUR and PyPi repo would need to be added to Bioarchlinux repo first in order for rnalysis to be added

| Dependency  | Package name | Source |
| ------------- | ------------- | ------------- |
numpy	| python-numpy |	Arch|
pandas	| python-pandas |	Arch|
scipy	| python-scipy |Arch|
matplotlib	| python-matplotlib | Arch|
requests | python-requests | Arch|
scikit-learn	| python-scikit-learn | Arch|
seaborn	| python-seaborn | Arch|
statsmodels | python-statsmodels | Arch|
joblib	| python-joblib | Arch|
tqdm	| python-tqdm | Arch|
appdirs	| python-appdirs | Arch|
pyyaml	| python-yaml | Arch|
typing_extensions | python-typing_extensions | Arch|
PyQt5	| python-pyqt5 | Arch|
qdarkstyle	| python-qdarkstyle | Arch|
R 	| r	| Arch|
graphviz 	| graphviz 	| Arch|
numba	| python-numba	| AUR|
hdbscan	| python-hdbscan	| AUR|
matplotlib-venn	| python-matplotlib-venn	| AUR|
Kallisto	| kallisto	| AUR|
scikit-learn-extra |	| pypi|
grid_strategy	|	| PyPi|
UpSetPlot		|	| PyPi|
xlmhg		|	| PyPi|
plotly (dependency for xlmhg) |python-plotly	| AUR|
pairwisedist	|	| PyPi|
cutadapt		|	| PyPi|

**Task List for new packages to be added to repo as dependencies**

- [x] #110
- [x] grid_strategy
- [x] UpSetPlot
- [x] #112
- [x] pairwisedist
- [x] #108
- [x] numba
- [x] hdbscan
- [x] matplotlib-venn
- [x] Kallisto

i have done preliminary work regarding writing `PKGBUILD` for cutadapt along with it dependencies. So far cutadapt seems to compile fine.

**Comment by starsareintherose** [comment]
Should it be closed?

**Comment by kbipinkumar** [comment]
> Should it be closed?

all the dependencies has been packaged properly. the actual RNAlysis package also compiles without issues but fails to launch. i am working with [upstream](https://github.com/GuyTeichman/RNAlysis/issues/9) to figure out what could be issue.

**Comment by kbipinkumar** [comment]
although GUI tests did successfully complete, some tests related to actual functionality seem to have failed. here is the truncated output below

```
=========================== short test summary info ============================
FAILED tests/test_clustering.py::test_clicom_cluster_wise_result - joblib.ext...
FAILED tests/test_clustering.py::test_clicom_get_cluster_similarity_matrix - ...
FAILED tests/test_clustering.py::test_clicom_inter_cluster_similarity[0-1-0.2222222222222222]
FAILED tests/test_clustering.py::test_clicom_inter_cluster_similarity[0-3-0.6666666666666666]
FAILED tests/test_clustering.py::test_clicom_inter_cluster_similarity[0-5-1]
FAILED tests/test_clustering.py::test_clicom_inter_cluster_similarity[3-5-0.6666666666666666]
FAILED tests/test_enrichment.py::test_enrichment_randomization_parallel_reliability
FAILED tests/test_enrichment.py::test_enrichment_parallel_validity - joblib.e...
FAILED tests/test_enrichment.py::test_enrich_hypergeometric_api - joblib.exte...
FAILED tests/test_enrichment.py::test_enrich_hypergeometric_pvalues - joblib....
FAILED tests/test_enrichment.py::test_go_enrichment_api[auto-hypergeometric-classic-kwargs0]
FAILED tests/test_enrichment.py::test_go_enrichment_api[auto-fisher-elim-kwargs1]
FAILED tests/test_enrichment.py::test_go_enrichment_api[caenorhabditis elegans-randomization-no-kwargs2]
FAILED tests/test_enrichment.py::test_kegg_enrichment_api[auto-hypergeometric-kwargs0]
FAILED tests/test_enrichment.py::test_kegg_enrichment_api[6239-fisher-kwargs1]
FAILED tests/test_enrichment.py::test_kegg_enrichment_api[caenorhabditis elegans-randomization-kwargs2]
FAILED tests/test_io.py::test_golr_annotation_iterator_golr_request_connectivity
===== 17 failed, 1439 passed, 1 skipped, 480 warnings in 503.53s (0:08:23) =====
```
i am attaching the complete [log](https://gist.github.com/kbipinkumar/17da7b9fc9f4c358951fa52fd1439505) here.

**Comment by kbipinkumar** [comment]
although GUI tests did successfully complete, tests related to actual functionality seem to have failed. here is the truncated output below

```
=========================== short test summary info ============================
FAILED tests/test_clustering.py::test_clicom_cluster_wise_result - joblib.ext...
FAILED tests/test_clustering.py::test_clicom_get_cluster_similarity_matrix - ...
FAILED tests/test_clustering.py::test_clicom_inter_cluster_similarity[0-1-0.2222222222222222]
FAILED tests/test_clustering.py::test_clicom_inter_cluster_similarity[0-3-0.6666666666666666]
FAILED tests/test_clustering.py::test_clicom_inter_cluster_similarity[0-5-1]
FAILED tests/test_clustering.py::test_clicom_inter_cluster_similarity[3-5-0.6666666666666666]
FAILED tests/test_enrichment.py::test_enrichment_randomization_parallel_reliability
FAILED tests/test_enrichment.py::test_enrichment_parallel_validity - joblib.e...
FAILED tests/test_enrichment.py::test_enrich_hypergeometric_api - joblib.exte...
FAILED tests/test_enrichment.py::test_enrich_hypergeometric_pvalues - joblib....
FAILED tests/test_enrichment.py::test_go_enrichment_api[auto-hypergeometric-classic-kwargs0]
FAILED tests/test_enrichment.py::test_go_enrichment_api[auto-fisher-elim-kwargs1]
FAILED tests/test_enrichment.py::test_go_enrichment_api[caenorhabditis elegans-randomization-no-kwargs2]
FAILED tests/test_enrichment.py::test_kegg_enrichment_api[auto-hypergeometric-kwargs0]
FAILED tests/test_enrichment.py::test_kegg_enrichment_api[6239-fisher-kwargs1]
FAILED tests/test_enrichment.py::test_kegg_enrichment_api[caenorhabditis elegans-randomization-kwargs2]
FAILED tests/test_io.py::test_golr_annotation_iterator_golr_request_connectivity
===== 17 failed, 1439 passed, 1 skipped, 480 warnings in 503.53s (0:08:23) =====
```
i am attaching the complete [log](https://gist.github.com/kbipinkumar/17da7b9fc9f4c358951fa52fd1439505) here.

**Comment by kbipinkumar** [comment]
> > Should it be closed?
> 
> all the dependencies has been packaged properly. the actual RNAlysis package also compiles without issues but fails to launch. i am working with [upstream](https://github.com/GuyTeichman/RNAlysis/issues/9) to figure out what could be issue.

although GUI tests did successfully complete, tests related to actual functionality seems to have failed. here is the truncated output below

```
=========================== short test summary info ============================
FAILED tests/test_clustering.py::test_clicom_cluster_wise_result - joblib.ext...
FAILED tests/test_clustering.py::test_clicom_get_cluster_similarity_matrix - ...
FAILED tests/test_clustering.py::test_clicom_inter_cluster_similarity[0-1-0.2222222222222222]
FAILED tests/test_clustering.py::test_clicom_inter_cluster_similarity[0-3-0.6666666666666666]
FAILED tests/test_clustering.py::test_clicom_inter_cluster_similarity[0-5-1]
FAILED tests/test_clustering.py::test_clicom_inter_cluster_similarity[3-5-0.6666666666666666]
FAILED tests/test_enrichment.py::test_enrichment_randomization_parallel_reliability
FAILED tests/test_enrichment.py::test_enrichment_parallel_validity - joblib.e...
FAILED tests/test_enrichment.py::test_enrich_hypergeometric_api - joblib.exte...
FAILED tests/test_enrichment.py::test_enrich_hypergeometric_pvalues - joblib....
FAILED tests/test_enrichment.py::test_go_enrichment_api[auto-hypergeometric-classic-kwargs0]
FAILED tests/test_enrichment.py::test_go_enrichment_api[auto-fisher-elim-kwargs1]
FAILED tests/test_enrichment.py::test_go_enrichment_api[caenorhabditis elegans-randomization-no-kwargs2]
FAILED tests/test_enrichment.py::test_kegg_enrichment_api[auto-hypergeometric-kwargs0]
FAILED tests/test_enrichment.py::test_kegg_enrichment_api[6239-fisher-kwargs1]
FAILED tests/test_enrichment.py::test_kegg_enrichment_api[caenorhabditis elegans-randomization-kwargs2]
FAILED tests/test_io.py::test_golr_annotation_iterator_golr_request_connectivity
===== 17 failed, 1439 passed, 1 skipped, 480 warnings in 503.53s (0:08:23) =====
```
i am attaching the complete [log](https://gist.github.com/kbipinkumar/17da7b9fc9f4c358951fa52fd1439505) here.

**Comment by kbipinkumar** [comment]
the package now successfully build and runs with suggested workarounds from upstream. the current version 3.3.0 still has some minor issues such as `setup.py` installing a tests folder to site-packages folder which is discouraged by Arch packaging guidelines. The upstream has acknowledged the issue and fix is incorporated for version 3.4.0 to be out in near future. 

once version 3.4.0 is out the package can be added to repository subject to successful builds
